### ISSUE TITLE
No logs when using --shell option

### ISSUE THREAD

**Issue by Frodox** [rpm-software-management/mock#17]
When I'm using `mock` like
```
mock --resultdir /myresults --shell "some command"
```
it doesn't store any logs in `/myresults` dir at all. I expect at least `build.log`, `root.log` and `state.log`. Because, in case of any errors with env (broken repo) it is hard to detect it without manually running this command...
Also, it would be cool to log runned commands in `state.log` (as with `--chroot` option), and also save `--shell` stdout somewhere may be :)

**Comment by msimacek** [comment]
Mock exlicitly disables logging when using `--shell`/`--chroot` in order to maintain backwards compatibility with existing tools, such as fedora-review, which expect this behavior and would break if it changed (yes, it sounds a bit silly, but fedora-review really breaks when `--shell` creates log files).
As you already noted, `--chroot` does log, so for your use case, you can use that instead, or capture mock output by ordinary shell redirection.

**Comment by xsuchy** [comment]
+1 to what @msimacek said. Just one note: 
   mock --chroot "some command"
is the same as --shell. You have to pass to --chroot at least two paramaters. It is documented in man page.

**Comment by Frodox** [comment]
> As you already noted, `--chroot` does log, so for your use case, you can use that instead, 
> `mock --chroot "some command"` is the same as `--shell`.

No and no. If it's *exactly* same, why is it not marked in man page? It's different, because `--chroot` doesn't split stdout and stderr as `--shell` does from chroot inside. So, I can't use it like
```bash
$ a=$(mock --chroot "ls" "listing" "boot")
INFO: mock.py version 1.2.21 starting (python version = 2.7.5)...
...
Start: chroot ['ls', 'listing', 'boot']
Finish: chroot ['ls', 'listing', 'boot']
ERROR: Command failed. See logs for output.
 # ls listing boot
```
because I will get all cmd's stderr as mock's stdout! It is weird.
```bash
$ echo "$a"
boot:
ls: cannot access 'listing': No such file or directory
```

And it works as expected with `--shell`.
```bash
$ b=$(mock --shell "ls listing boot")                                                                               
INFO: mock.py version 1.2.21 starting (python version = 2.7.5)...
...
Start: shell
ls: cannot access 'listing': No such file or directory
Finish: shell

$ echo "$b"
boot:
```

> or capture mock output by ordinary shell redirection.

No, I can't :-) Because `mock` change it's output depends on redirection... so, for example, I have problem with repos and run `mock --shell` **at first time**.
```bash
$ mock --shell "cmd"
...
Could not retrieve mirrorlist http://mmirrors.fedoraproject.org/mirrorlist?repo=rawhide&arch=x86_64 error was
14: curl#6 - "Could not resolve host: mmirrors.fedoraproject.org; Name or service not known"

 One of the configured repositories failed (Unknown),
 and yum doesnt have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.
...

$ mock --shell "cmd" 1>stdout 2>stderr
$ cat stdout
(empty)
$ cat stderr
INFO: mock.py version 1.2.21 starting (python version = 2.7.5)...
...
Start: yum install
ERROR: Command failed. See logs for output.
 # /usr/bin/yum --installroot /var/lib/mock/fedora-rawhide-x86_64...
```
and thats all...

So, I can't detect such error in logs on the host. And I need to manually run the command and see problems =( It is my pain..

> Mock exlicitly disables logging when using --shell/--chroot in order to maintain backwards compatibility with existing tools, such as fedora-review, which expect this behavior and would break if it changed (yes, it sounds a bit silly, but fedora-review really breaks when --shell creates log files).

@msimacek @xsuchy Do you think, it is really correct? It looks like dirty hack for me :) 
I suggest, at least, add NOTE to man page, that `--shell` disables any logs. 
And how about another way - implement mock's `--no-logs` option, and fix `fedora-review` to call `mock --no-logs  --shell "..."` with this option ? It looks like architectural right decision, doesn't it?

**Comment by xsuchy** [comment]
> because I will get all cmd's stderr as mock's stdout! It is weird.

Yes. But originally you asked for the log and you stated that there is no log. With --chroot you get the log. Yes, the stdout and stderr is mixed, there is place for improvement.

> I suggest, at least, add NOTE to man page

Done in d02d30b

> And how about another way - implement mock's --no-logs option. 

I agree that the current implementation of logging is not ideal. So any contribution is welcome. Just be aware that it must work with --new-chroot too.

**Comment by Frodox** [comment]
> Yes, the stdout and stderr is mixed, there is place for improvement.

Are there issues for that? Or should I create new one? :)

> Done in d02d30b

Thank you!

> I agree that the current implementation of logging is not ideal. So any contribution is welcome. Just be aware that it must work with --new-chroot too.

Ok. I'll keep in mind.

**Comment by xsuchy** [comment]
> Are there issues for that? Or should I create new one? :)

We can keep this one :)
